### ISSUE TITLE
jsonlint skips duplicate keys

### ISSUE THREAD

**Issue by cilindrox** [github/super-linter#1737]
## Describe the bug

JSON lint checks should fail when there's a duplicate key on a JSON object. Currently, the `jsonlint` linter won't report any issues.

## Expected behavior

Duplicate key entries should raise an error or at least the behavior should be configurable/raise a warning.
Per #1524 eslint might be preferred to the `jsonlint` linter, but I couldn't figure out how to replace it even after copying verbatim the `.eslintrc.yml` template into `.github/linters` on the repo.

## Steps to Reproduce

Create a test file `bad.json` with a duplicate `foo` key:

```bash
cat >bad.json<<EOF
{
    "foo": "bar",
    "baz": "quux",
    "foo": "quux"
}
EOF
```

Run the linter:

```bash
docker run \
  -e RUN_LOCAL=true \
  -e ACTIONS_RUNNER_DEBUG=true \
  -e DISABLE_ERRORS=false \
  -e ERROR_ON_MISSING_EXEC_BIT=true \
  -e LINTER_RULES_PATH=. \
  -e MULTI_STATUS=false \
  -e VALIDATE_ALL_CODEBASE=true \
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:slim-v4
```

yields:

```
2021-07-08 18:56:05 [DEBUG]   Linting FILE: /tmp/lint/bad.json
2021-07-08 18:56:05 [DEBUG]   FILE_STATUS (bad.json) is empty, or not set to 'good' or 'bad'. Assuming it should be linted correctly. Setting FILE_STATUS to 'good'...
2021-07-08 18:56:05 [DEBUG]   File: /tmp/lint/bad.json, FILE_NAME: bad.json, DIR_NAME:/tmp/lint, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: json
2021-07-08 18:56:05 [INFO]   ---------------------------
2021-07-08 18:56:05 [INFO]   File:[/tmp/lint/bad.json]
2021-07-08 18:56:05 [INFO]    - File:[bad.json] was linted with [jsonlint] successfully
2021-07-08 18:56:05 [DEBUG]   Error code: 0. Command output:
------
{
  "foo": "quux",
  "baz": "quux"
}
------
```

(notice the output is "prettied", the last entry of `foo` overrides the initial one)

## Additional context

It could be argued that although [the spec doesn't really mention](https://stackoverflow.com/a/21833017/992453) whether duplicates are invalid, these seem like a common-enough source of errors and undesired behavior. As a matter of fact, [there's a 10-year old bug on the linter repo](https://github.com/zaach/jsonlint/issues/13) arguing just this.

**Comment by admiralAwkbar** [comment]
@cilindrox Hey this is a good catch.

Currently we use `jsonlint` for pure `.json` files but we use `eslint` for `.jsonc` and `.json5`. 

I think we could look at trying to remove `jsonlint` and only use `eslint` to handle this and see if we get better behavior

Will push a try at this now :)

**Comment by cilindrox** [comment]
@admiralAwkbar that'd be super nice - initially [I thought this was the case](https://github.com/github/super-linter/pull/1524) (`jsonlint` being deprecated in favor of `eslint`) but wasn't able to make it work. Having these flagged at least as a warn would be incredibly helpful :)
